runtime/dancebox/src/migrations.rs

```diff
@@ -108,15 +111,117 @@ where
     }
 }
 
+pub struct MigrateHoldReason<T>(pub PhantomData<T>);
+impl<T> Migration for MigrateHoldReason<T>
+where
+    T: pallet_balances::Config,
+    T: pallet_pooled_staking::Config,
+    T::HoldIdentifier: From<crate::HoldReason>,
+{
+    fn friendly_name(&self) -> &str {
+        "TM_MigrateHoldReason"
+    }
+
+    fn migrate(&self, _available_weight: Weight) -> Weight {
+        log::info!(target: LOG_TARGET, "migrate");
+        let pallet_prefix: &[u8] = b"Balances";
+        let storage_item_prefix: &[u8] = b"Holds";
+
+        let stored_data: Vec<_> = storage_key_iter::<
+            T::AccountId,
+            BoundedVec<IdAmount<[u8; 8], <T as pallet_balances::Config>::Balance>, T::MaxHolds>,
+            Blake2_128Concat,
+        >(pallet_prefix, storage_item_prefix)
+        .collect();
+
+        let migrated_count_read = stored_data.len() as u64;
+        let mut migrated_count_write = 0u64;
+
+        // Write to the new storage
+        for (account_id, holds) in stored_data {
+            let mut new_holds = vec![];
+
+            for hold in holds {
```

Comment:
Would that turn all holds into PooledStake holds? No other pallets perform holds?

Reply:
we only have one hold type yes. The problem if I evaluate the hold type is that I still need to convert the rest of the hold identifiers to something and that something right now is only PooledStake